exotica/src/Scene.cpp

```diff
@@ -840,9 +845,55 @@ namespace exotica
   void Scene::cleanScene()
   {
       getPlanningScene()->removeAllCollisionObjects();
+      updateSceneFrames();
       collision_scene_->reinitializeCollisionWorld();
   }
 
+  void Scene::updateSceneFrames()
+  {
+      kinematica_.resetModel();
+
+      planning_scene::PlanningScenePtr ps = collision_scene_->getPlanningScene();
+
+      // Add world objects
+      for(auto& object : *ps->getWorld())
+      {
+          if(object.second->shapes_.size())
+          {
+              // Use the first collision shape as the origin of the object
+              Eigen::Affine3d objTransform = object.second->shape_poses_[0];
+              std::string objectName = object.first;
```

Comment:
``&`` reference?

Reply:
Where? ```object.first``` is a ```std::string```.

Reply:
``std::string& objectName = object.first``

Reply:
That stuff gets optimized out anyway.

Reply:
Ok. I admit it. It doesn't. I have removed the whole variable.